### PR TITLE
Remove NeogitCommitMessage filetype

### LINKED ISSUE #405: Use git commit file type for commits
**Is your feature request related to a problem? Please describe.**
Neovim has a git commit file type. Several plugins, like cmp-git use this. However, neogit uses its own git file type, preventing cmp-git and others from working.

**Describe the solution you'd like**

Neogit should use the built-in file type 
**Describe alternatives you've considered**
Adjust other plugins to use the neogit type. This seems inefficient.

### PR DESCRIPTION
**Pull request by vinibispo**
Close #405 
- Remove NeogitCommitMessage filetype
- Turn it back to gitcommit filetype

### CODE REVIEW

File: syntax/NeogitStatus.vim [hunk 1]
```diff
@@ -5,9 +5,9 @@ endif
 source $VIMRUNTIME/syntax/gitrebase.vim
 
 syn match NeogitObjectId /^[a-z0-9]\{7,}\>/
-syn match NeogitCommitMessage /.*/ contained
-syn match NeogitBranch /\S\+/ contained nextgroup=NeogitCommitMessage
-syn match NeogitRemote /\S\+/ contained nextgroup=NeogitCommitMessage
+syn match gitcommit /.*/ contained
+syn match NeogitBranch /\S\+/ contained nextgroup=gitcommit
+syn match NeogitRemote /\S\+/ contained nextgroup=gitcommit
```
Comment (gegoune) [inline syntax/NeogitStatus.vim:10]: I am not entirely sure but those probably can just be removed. Nvim comes with gitcommit syntax as well as there is treesitter parser for gitconmit now.

Comment (vinibispo) [reply syntax/NeogitStatus.vim:10]: I'll try removing

Comment (CKolkey) [reply syntax/NeogitStatus.vim:10]: Yea, these can be removed altogether.

**Comment by CKolkey** [comment]
This change would prevent `lua/neogit/autocmds.lua:19` from working properly. Not sure the consequences of that.

**Comment by gegoune** [comment]
```lua
      if api.nvim_buf_get_option(o.buf, "filetype"):find("Neogit") then
        return
      end
```
This could/should also check for `gitcommit` in that case. Not sure what implications of not checking that are, refreshing Neogit on`gitcommit` buffers' save makes little sense here I think.

**Comment by CKolkey** [comment]
Yea. I've been running with this patch for a week or two now without issue, and without addressing the above autocmd.

File: syntax/NeogitStatus.vim (hunk 1 above)
Comment (CKolkey) [reply syntax/NeogitStatus.vim:10]: An update on this: These should _not_ be removed, and should be left unchanged. I was mistaken before: they have nothing to do with the NeogitGitCommit popup. These three lines are just for syntax highlighting on the status buffer of the branch name/remote name.

**Comment by vinibispo** [comment]
Sorry guys, I'll promise that I'll fix it

**Pull request closed without merging**